app/post.go

```diff
@@ -632,3 +658,14 @@ func GetOpenGraphMetadata(url string) *opengraph.OpenGraph {
 
 	return og
 }
+
+func UserHasPermissionToExecute(userId string, permission *model.Permission) bool {
```

Comment:
You shouldn't need this function any more if you consolidate checking the setting above. I also don't think we would want a function like this since we should encourage the caller to get the user themselves to avoid duplicate database hits